plugins/web-terminal/src/components/TerminalComponent/TerminalComponent.tsx

```diff
@@ -1,27 +1,117 @@
-import { makeStyles } from '@material-ui/core';
-import React from 'react';
+import { Button, makeStyles, TextField } from '@material-ui/core';
+import React, { useCallback, useRef } from 'react';
 import { Terminal } from 'xterm';
 import { FitAddon } from 'xterm-addon-fit';
+import { AttachAddon } from 'xterm-addon-attach';
+import { InfoCard, Progress } from '@backstage/core-components';
+import { useApi, configApiRef } from '@backstage/core-plugin-api';
+
 import './static/xterm.css';
 
 const useTerminalStyles = makeStyles({
   term: {
     width: '100%',
-    height: '100%',
+    height: '400px',
+  },
+  formDisplay: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    flexDirection: 'column',
+    rowGap: '10px',
+    margin: '10px',
+  },
+  centerInfo: {
+    textAlign: 'center',
+    marginTop: '10px',
+    marginBottom: '10px',
   },
 });
+
+interface IFormData {
+  token: string | undefined;
+  cluster: string | undefined;
+}
+
 export const TerminalComponent = () => {
-  const fitAddon = new FitAddon();
-  const terminal = new Terminal();
-  terminal.loadAddon(fitAddon);
-  fitAddon.fit();
+  const [websocketRunning, setWebsocketRunning] = React.useState(false);
+  const [loading, setLoading] = React.useState(false);
+  const [formData, setFormData] = React.useState<IFormData>({
+    token: undefined,
+    cluster: undefined,
+  });
   const classes = useTerminalStyles();
+  const tokenRef = React.useRef<HTMLInputElement>(null);
+  const clusterRef = React.useRef<HTMLInputElement>(null);
+  const termRef = useRef(null);
```

Comment:
Just a nit, using both `React.useRef` and `useRef` looks a bit messy, let's use just one.. either is fine.

Reply:
Resolved in latest commit